### PR TITLE
chore: add a safeguard for setting server address through the omnibar

### PR DESCRIPTION
**Pull request by hamidzr**
## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Review by apizzini — APPROVED** [review]
Looks good.

Didn't know about the omnibar.

2 nit thing:
- close the omnibar after the command is submitted?
- when setting the dev server we could potentially reload the page (need to do that manually)?

**Comment by hamidzr** [comment]
> close the omnibar after the command is submitted?

there has been some talks around this one. There is a google document and a jira ticket we're using to collect feedback on it and this is one of them 🤔  (I'll share the links)

**Comment by hamidzr** [comment]
> when setting the dev server we could potentially reload the page (need to do that manually)?

I meant to send the user to the login page but I'm now sending users to the dashboard. I think that should be enough and no reload would be necessary

**Comment by hamidzr** [comment]
I made two changes:
- send the user to the logout route after a server address change
- treat server address reset as a server address change

File: webui/react/src/dev.ts [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+import { paths, routeToReactUrl, serverAddress } from 'routes/utils';
 import * as Api from 'services/api';
 import { updateDetApi } from 'services/apiConfig';
 
+const onServerAddressChange = () => {
+  updateDetApi({ basePath: serverAddress() });
+  routeToReactUrl(paths.logout());
+};
+
 export const setServerAddress = (address: string): void => {
```
Comment (hamidzr) [inline webui/react/src/dev.ts:12]: I"m not adding the same safeguard here to allow for more flexibility when going this route.

**Pull request merged**